test/async-hooks/test-timerwrap.setInterval.js

```diff
@@ -51,6 +51,6 @@ function onexit() {
   hooks.disable();
   hooks.sanityCheck('TIMERWRAP');
 
-  checkInvocations(t, { init: 1, before: 3, after: 3, destroy: 1 },
+  checkInvocations(t, { init: 1, before: 3, after: 3 },
```

Comment:
This seems wrong. Is it from the async-hooks place I noted previously?

Reply:
There's a single TimerWrap, there's no real opportunity to destroy it before exiting.

Reply:
Won't something with a `uv_before` handle work? I think an immediate should do the trick?

Reply:
> I'm still a little confused as to why handle cleanup doesn't happen this turn of the event loop, though?

There's one global handle for the duration of the process. The goal is to eventually have this be an Environment bound uv_timer_t much like Immediates, rather than a HandleWrap. (See my other comment.)

By the time the process is exiting, it's too late to cleanup handles because they take one event loop run to fully clear plus destroy hooks won't be emitted anyway because they run in an immediate.

Reply:
But there's nothing to delay for in this case. The handle doesn't get destroyed during the duration of the process. It starts on the first timer run and then exists until the process exits.

(This test is going away when TimerWrap goes.)

Reply:
Feel free to have a look at for the follow-up that removes TimerWrap altogether: https://github.com/apapirovski/node/commit/d8f075da7b8129f59b1d537b6475394425b81c3e